### ISSUE TITLE
Can not play audio in background (IOS)

### ISSUE THREAD

**Issue by hulastone** [matthew-carroll/fluttery_audio#10]
Version ^0.0.3 and ^0.0.4 can not play audio music (.m4a) in IOS backgroud mode, on real device but in simulator is ok. Please help to fix. Thanks so much.

**Comment by hulastone** [comment]
i solved the problem 1 (can not play background on ios) with below help topic.
https://stackoverflow.com/questions/3745067/how-to-play-background-audio-using-avfoundation-for-iphone-ipoad-touch-with-ios

Thanks for contribute this plugin.

**Comment by hulastone** [comment]
about problem #2 (error when install on ios real device)
i worked around by comment below code 
- (void) seek:(int64_t) seekPositionInMillis {
  NSLog(@"%@: %@: %lli", TAG, @"seek()", seekPositionInMillis);
  CMTime seekTime = CMTimeMakeWithSeconds(seekPositionInMillis / 1000, NSEC_PER_SEC);
  NSLog(@"%@: Decoded seek seconds: %f", TAG, CMTimeGetSeconds(seekTime));
  
  for (id<AudioPlayerListener> listener in [_listeners allObjects]) {
    [listener onSeekStarted];
  }
  /*
  [_audioPlayer seekToTime:seekTime completionHandler:^(bool finished) {
    for (id<AudioPlayerListener> listener in [self->_listeners allObjects]) {
      [listener onSeekCompleted:[self playbackPosition]];
    }
  }];
  */
}

**Comment by chrislong365** [comment]
for #1 I added this code to the ios AudioPlayer.m file (/ios/Classes/AudioPlayer.m)
After     [_audioPlayer play];

`    BOOL sucess = [[AVAudioSession sharedInstance]
                    setCategory:AVAudioSessionCategoryPlayback
                  error:nil];`



Not sure if it is the correct Swift way but it worked.

**Comment by hulastone** [comment]
yes, sure i tried before. But it's can not move to next track in background mode.

**Comment by chrislong365** [comment]
might need to release AVAudioSession at stop or next? I'm not sure how that would work though.

**Comment by hulastone** [comment]
i saw the reference: https://stackoverflow.com/questions/9660488/ios-avaudioplayer-doesnt-continue-to-next-song-while-in-background

but i haven't tried yet. Maybe it's OK.

**Comment by prasantd** [comment]
There is a forked branch in fluttery_audio that solves my issues of playing it in iOs - play - pause issue and playing in backgrounding ios. The question is how do I use that because the pull request has not been merged yet in the published version?
